### PR TITLE
Qt Offroad stats

### PR DESCRIPTION
**Pull request by ghost**
Add driving stats to offroad main window in Qt

### CODE REVIEW

**Comment by jahanzeeb** [comment]
Ok

**Review by Torresangl925 — APPROVED** [review]

File: selfdrive/ui/qt/widgets/drive_stats.cc [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+  int all_distance = all["distance"].toDouble()*(metric ? MILE_TO_KM : 1);
+  int all_minutes = all["minutes"].toDouble();
+  int all_routes = all["routes"].toDouble();
+  int week_distance = week["distance"].toDouble()*(metric ? MILE_TO_KM : 1);
+  int week_minutes = week["minutes"].toDouble();
+  int week_routes = week["routes"].toDouble();
+
+  QGridLayout *gl = new QGridLayout();
+
+  QLabel *past_week = new QLabel(bold("PAST WEEK"));
```
Comment (pd0wm) [inline selfdrive/ui/qt/widgets/drive_stats.cc]: Remove duplication. Extract into function.

File: selfdrive/ui/qt/widgets/drive_stats.cc [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+  QString token = create_jwt(dongle_id);
+
+  QNetworkAccessManager *manager = new QNetworkAccessManager(this);
+  connect(manager, &QNetworkAccessManager::finished, this, &DriveStats::replyFinished);
+
+  QNetworkRequest request;
+  request.setUrl(QUrl("https://api.commadotai.com/v1.1/devices/" + dongle_id + "/stats"));
+  request.setRawHeader("Authorization", ("JWT "+token).toUtf8());
+
+  manager->get(request);
```
Comment (pd0wm) [inline selfdrive/ui/qt/widgets/drive_stats.cc:196]: periodically refresh

**Comment by Liverpoolj** [comment]
Hey am new to this and don't understand much of so any help would be much appreciated

**Review by Kamil31989 — APPROVED** [review]

**Pull request merged**